File: VB/CustomWizardExample/Wizard/CustomDataSourceWizardPageFactory.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports DevExpress.DataAccess.UI.Wizard
Imports DevExpress.DataAccess.Wizard
Imports DevExpress.DataAccess.Wizard.Presenters
Imports DevExpress.DataAccess.Wizard.Views

Namespace CustomWizardExample.Wizard
	Public Class CustomDataSourceWizardPageFactory
		Inherits WizardPageFactory(Of Model.CustomReportModel, IDataSourceWizardClient)

		Public Sub New(ByVal client As IDataSourceWizardClient)
			MyBase.New(client)
		End Sub


        Protected Overrides Sub RegisterDependencies(client As IDataSourceWizardClient)
            MyBase.RegisterDependencies(client)

            Container.RegisterType(Of ChooseDataSourceTypePage(Of Model.CustomReportModel), Presenters.ChooseDataSourceTypePageEx)()

            Container.RegisterType(Of IChooseDataSourceTypePageView, Views.ChooseDataSourceTypePageViewEx)()

            Container.RegisterType(Of Presenters.ChoosePersonsDataSourceTypePage, Presenters.ChoosePersonsDataSourceTypePageEx)()
            Container.RegisterType(Of Views.IChoosePersonsDataSourcePageView, Views.ChoosePersonsDataSourcePageView)()
        End Sub
	End Class
End Namespace

File: CS/CustomWizardExample/Wizard/CustomDataSourceWizardPageFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.DataAccess.UI.Wizard;
using DevExpress.DataAccess.Wizard;
using DevExpress.DataAccess.Wizard.Presenters;
using DevExpress.DataAccess.Wizard.Views;

namespace CustomWizardExample.Wizard {
    public class CustomDataSourceWizardPageFactory : WizardPageFactory<Model.CustomReportModel, IDataSourceWizardClient> {
        public CustomDataSourceWizardPageFactory(IDataSourceWizardClient client) : base(client) { }


        protected override void RegisterDependencies(IDataSourceWizardClient client) {
            base.RegisterDependencies(client);

            Container.RegisterType<ChooseDataSourceTypePage<Model.CustomReportModel>, Presenters.ChooseDataSourceTypePageEx>();

            Container.RegisterType<IChooseDataSourceTypePageView, Views.ChooseDataSourceTypePageViewEx>();

            Container.RegisterType<Presenters.ChoosePersonsDataSourceTypePage, Presenters.ChoosePersonsDataSourceTypePageEx>();
            Container.RegisterType<Views.IChoosePersonsDataSourcePageView, Views.ChoosePersonsDataSourcePageView>();
        }
    }
}
